sql/importer: TestRandomParquetExports failed

sql/importer.TestRandomParquetExports [failed](https://mesolite.cluster.engflow.com/invocations/default/b11609db-181b-4c58-9d6d-720ef5a069e6?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2ltcG9ydGVyOmltcG9ydGVyX3Rlc3Q=) on release-24.1 @ [537e15cfb0c6abc49f88c72eca826a343093a1da](https://github.com/cockroachdb/cockroach/commits/537e15cfb0c6abc49f88c72eca826a343093a1da):


```
        	col4_3 BOX2D NOT NULL,
        	"c%beol4_4" FLOAT8 NOT NULL AS (abs(c😈ol4_2)) STORED,
        	" col4_5" FLOAT8 NOT NULL AS (abs(c😈ol4_2)) VIRTUAL,
        	col4_6 STRING NULL AS (lower("%pcol4_0")) VIRTUAL,
        	CONSTRAINT "ta̔b le_4_pkey" PRIMARY KEY ("😑coL4_1" ASC, " col4_5" ASC, col4_3 ASC),
        	INDEX "ta̔b le_4_col4_6_ col4_5_c😈ol4_2_expr_idx" (col4_6 ASC, " col4_5" ASC, c😈ol4_2 DESC, lower("%pcol4_0") ASC) WHERE (c😈ol4_2 > 1.401298464324817e-45:::FLOAT8) AND ("%pcol4_0" <= '':::STRING),
        	INDEX "ta̔b le_4_😑coL4_1_c😈ol4_2_idx" ("😑coL4_1" ASC, c😈ol4_2 ASC) STORING ("%pcol4_0") PARTITION BY LIST ("😑coL4_1", c😈ol4_2) (
        		PARTITION "ta̔b le_4_par%pt🙄_0" VALUES IN ((B'010011101001100011010011001010011000010010000100110111110110111101010100000111101000111010001000', -0.46726465245725257), (B'01010011000001000100001000001', -1.6269593674600804), (B'100010100001001101001001101100001011', -0.02117335773323785), (B'1011011110110100001011011111011010101100100100001111010100101111011111111111010111111001010101', 0.0781099643999178), (B'00001010000', 1.3950615847788295), (B'110100001010011100', -2.384868744119317)), 
        		PARTITION "%qt a̔b le_4_pa rt_1" VALUES IN ((B'00111110001101011010000000100101001100010110110111111011000110011100000110010', -0.5194767531961979), (B'10001011100010001011110011110000000001111100101', -2.2649094028638785), (B'10000010110101110001001100000000111000111001110010100100000101000000011010111010100110001011000011', 0.031216239427381942), (B'10001111110011110001000101010101110011011011000001000010110010110000010', 0.8677457486962883), (B'001111100', -0.25975079808729196), (B'1000100001111010011000110101101001011010110010111000010111010101100111101111111001010', -0.301466469263472)), 
        		PARTITION "ta̔b le_4_part_2" VALUES IN ((B'000011010101111010110000010000111100101011101100111100010000110001100110101100100', -0.7945003258138975), (B'110100100100110111100001100011010101', 0.840661645076534), (B'0000000000000000000000000000000000000000000000000000000000000000', -0.20567225769146874), (B'001100110001010101101111010', 0.5505145117520222), (B'1001001101111000011100010111100111010111000110010101100101000100000110010111', 0.6688528579268586), (B'01001100001011000000001000', 1.0)), 
        		PARTITION "ta̔b le_4_p art_3" VALUES IN ((B'00110110110010011101010111011000011101000001101000010110110101100110100', -0.9589465095170344), (B'000101011111110011001110011001010011110110101101111101110000100101010', 0.4719287622323405), (B'11000110000001', -0.008203574419113568), (B'11100001101001010010011', 0.08881098518607122), (B'11100', -0.6326894987202039), (B'0101100110011111110000111', -0.20564933785539918)), 
        		PARTITION "ta̔b le_4_part_4" VALUES IN ((B'1001100', -0.6849868758748617), (B'01010011000010111001101110000111101000100100', 0.0), (B'10100010001101001011000000101011000001011100010110101100101100110100111110001001001010100', 0.8012962440178721), (B'0111111111111111111111111111111111111111111111111111111111111111', -0.5660780402307181), (B'010111011100111111000000111100001010101011101010001110001000000110011011', 0.09882467978291759), (B'0101111011101111', 0.4019367110693842)), 
        		PARTITION "ta̔b le_4_part_5" VALUES IN ((B'11100111100100', 0.8257400958091845), (B'010010010111001011010011110011000000011010', 1.211308802456921), (B'00011101011101111101111', 0.5791776353001535), (B'0000000000000000000000000000000000000000000000000000000000000000', -0.48829145610300095), (B'0111111111111111111111111111111111111111111111111111111111111111', 0.5564744021903177), (B'001100100110010011001100111000100110111001111010111001101', -1.0663138851919607)), 
        		PARTITION "ta̔b le_4_part_6" VALUES IN ((B'11000000', -0.21937204591892412), (B'1010101010010011110100000011110001100110111', 3.4028234663852886e+38), (B'101011110011011111010', -0.9699886815009597), (B'011000010100010111110010001100011010001100010000110010100101011101000000100', 0.24070285069666775), (B'1110001101101011001011100110101000011', 0.284758308823967), (B'0111111111111111111111111111111111111111111111111111111111111111', 1.489618017471095)), 
        		PARTITION "ta̔b le_4_pa%qr t_7" VALUES IN ((B'1101000111110100000100111100110101', -0.07217715401419372), (B'01011110111000001011000011011011110110110', -1.0925552122756716), (B'001101110010011001101000010001101101111011', -1.202139228349786), (B'100000011010101', 0.9315700386768547), (B'00011110010100001100001100100101001101101110110101011010111000111100000', -0.28947655258901395), (B'0100000100011100000101001001101100001101', -0.1407639604438321)), 
        		PARTITION "ta̔b le_4_part_8" VALUES IN ((B'111111101011110011000101110010110111101010110001010100011101111001000100110111', -0.9018028615325585), (B'1110111111111011000011100010010010001', 1.248363982584444), (B'100001001001010011000011011100000110111001010010', 0.1156418589595698), (B'00111001111001001111110110011110100', -0.0), (B'111110101011110111110110101011010000011110001100010010', 0.21766233960003817), (B'011110010011001110001011110010011', -0.7807064341984993))
        	),
        	UNIQUE INDEX "ta̔b le_4_%pcol4_0_col4_3_c😈ol4_2_key" ("%pcol4_0" DESC, col4_3 ASC, c😈ol4_2 ASC),
        	INDEX "ta̔b le_4_expr_idx" (lower("%pcol4_0") DESC) WHERE (((("c%beol4_4" = '+Inf':::FLOAT8) OR ("%pcol4_0" < '':::STRING)) OR (c😈ol4_2 > (-1.0):::FLOAT8)) OR (" col4_5" >= 'NaN':::FLOAT8)) AND (col4_6 >= 'X':::STRING) VISIBILITY 0.08,
        	INDEX "ta̔b le_4_expr_idx1" (lower(CAST("😑coL4_1" AS STRING)) DESC) WHERE ((("%pcol4_0" <= e'\'':::STRING) AND ("c%beol4_4" >= 'NaN':::FLOAT8)) AND (" col4_5" >= 3.4028234663852886e+38:::FLOAT8)) AND (col4_6 > e'\x00':::STRING),
        	UNIQUE INDEX "ta̔b le_4_😑coL4_1_c😈ol4_2_key" ("😑coL4_1" ASC, c😈ol4_2 ASC) STORING ("c%beol4_4"),
        	INDEX "ta̔b le_4_c%beol4_4_😑coL4_1_ col4_5_idx" ("c%beol4_4" DESC, "😑coL4_1" ASC, " col4_5" ASC) STORING ("%pcol4_0", c😈ol4_2) VISIBILITY 0.03,
        	FAMILY "fam_0_c%beol4_4" ("c%beol4_4"),
        	FAMILY "fam_1_c😈ol4_2_😑coL4_1" (c😈ol4_2, "😑coL4_1"),
        	FAMILY "fam_2_%pcol4_0" ("%pcol4_0"),
        	FAMILY fam_3_col4_3 (col4_3)
        )
        -- Warning: Partitioned table with no zone configurations.
        ]]
    exportparquet_test.go:173: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql/importer_test/pkg/sql/importer/exportparquet_test.go:173
        	            				github.com/cockroachdb/cockroach/pkg/sql/importer_test/pkg/sql/importer/exportparquet_test.go:126
        	            				github.com/cockroachdb/cockroach/pkg/sql/importer_test/pkg/sql/importer/exportparquet_test.go:268
        	Error:      	Not equal: 
        	            	expected: &tree.DBitArray{BitArray:bitarray.BitArray{words:[]uint64{}, lastBitsUsed:0x0}}
        	            	actual  : &tree.DBitArray{BitArray:bitarray.BitArray{words:[]uint64(nil), lastBitsUsed:0x0}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -2,4 +2,3 @@
        	            	  BitArray: (bitarray.BitArray) {
        	            	-  words: ([]uint64) {
        	            	-  },
        	            	+  words: ([]uint64) <nil>,
        	            	   lastBitsUsed: (uint8) 0
        	Test:       	TestRandomParquetExports
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestRandomParquetExports3473253499
--- FAIL: TestRandomParquetExports (17.84s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #148581 sql/importer: TestRandomParquetExports failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.1.8-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomParquetExports.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-54722

--- a/pkg/sql/export/exportparquet_test.go
+++ b/pkg/sql/export/exportparquet_test.go
@@ -229,6 +229,22 @@ func TestRandomParquetExports(t *testing.T) {
 					require.NoError(t, err)
 
 					for _, col := range cols {
+						// Skip tables with BIT(0) columns (or arrays of BIT(0)).
+						// The testutils parquet decoder cannot decode these as
+						// into the expected empty bit array and decodes the as
+						// nil. Ignore these to avoid these flakes.
+						colTyp := col.Typ
+						if colTyp.Family() == types.ArrayFamily {
+							colTyp = colTyp.ArrayContents()
+						}
+						if colTyp.Family() == types.BitFamily && colTyp.Width() == 0 {
+							// We skip these because of the aforementioned
+							// decoder issue.
+							t.Logf("skipping table '%s' because it has BIT(0) column '%s' (type: %s)",
+								tableName, col.Name, colTyp.String())
+							return fmt.Errorf("table has BIT(0) column")
+						}
+
 						// TODO(#104278): don't call this function to check if a type is supported.
 						// We should explicitly use the ones supported by  util/parquet).
 						_, err := parquet.NewSchema([]string{"test"}, []*types.T{col.Typ})

